### DISCUSSION TITLE
Clarification regarding input privacy in SMPC video

### DISCUSSION THREAD

**Discussion by jaintj95** [OpenMined/courses#153] [Course: Our Privacy Opportunity] [answered]
Lesson 4: Secure Multi-Party Computation for Input Privacy 

Towards the end of the video (8:23 onwards), Andrew mentions: 
"Multiple people can encrypt their information, run it through a program and generate an output that only a third specific person can decrypt. That is input privacy"

Does input privacy also cover output of a computation?

**Comment by Nilanshrajput** [comment, marked as answer]
Hey @jaintj95, yes `Input Privacy` also covers the output of a flow/computation, `Input Privacy` ensures the privacy of inputs, intermediate states as well as output( with a case in which specific output could be designed for the particular owner), or you can think `Input privacy` ensures that it protects the complete pipe used for the flow of information and also make sure that the output is only seen by the specific person it was meant to be, but this should now be compared to `Output Privacy` as this doesn't cover what information can that output leak about the input to the person who can access the output and that's the thing to differentiate from `Output Privacy`.
Now in case of SMPC, the output is decrypted only to a specific person, you could think of an example as encrypted ML training, where the model and input both are encrypted using SMPC, now the output will only be decrypted from the model owner.
